DigitalLearningSolutions.Data/Services/UserService.cs

```diff
@@ -73,5 +76,34 @@ public List<CentreUserDetails> GetUserCentres(AdminUser? adminUser, List<Delegat
 
             return availableCentres.OrderByDescending(ac => ac.IsAdmin).ThenBy(ac => ac.CentreName).ToList();
         }
+
+        public bool TryUpdateUserAccountDetails(int? adminId, int? delegateId, string password, string firstName, string surname, string email)
+        {
+            var (loggedInAdminUser, loggedInDelegateUser) = GetUsersById(adminId, delegateId);
+
+            var signedInEmail = loggedInAdminUser?.EmailAddress ?? loggedInDelegateUser?.EmailAddress;
+
+            var (adminUser, delegateUsers) = GetUsersByEmailAddress(signedInEmail);
+
+            var (verifiedAdminUser, verifiedDelegateUsers) =
+                loginService.VerifyUsers(password, adminUser, delegateUsers);
+
+            if (verifiedAdminUser == null && verifiedDelegateUsers.Count == 0)
+            {
+                return false;
+            }
+
+            if (verifiedAdminUser != null)
+            {
+                userDataService.UpdateAdminUser(firstName, surname, email, verifiedAdminUser.Id);
+            }
+
+            if (verifiedDelegateUsers.Count != 0)
+            {
+                userDataService.UpdateDelegateUsers(firstName, surname, email, verifiedDelegateUsers.Select(d => d.Id).ToArray());
```

Comment:
This looks good, but could we pull the mapping to ids into a variable above this line for better readability?